### ISSUE TITLE
feature: password protected trades/offers

### ISSUE THREAD

**Issue by LeoSpyke** [RoboSats/robosats#714]
**Is your feature request related to a problem? Please describe.**
Sometimes it might happen that two actors want to perform a "private" trade, with pre-agreed conversion rates and currencies, but without giving up the functionality and "insurance" service offered by a third-party escrow.

**Describe the solution you'd like**
The idea is to simply add a "Private trade" checkbox during the "Sell Order" creation which enables a password input field.

**Describe alternatives you've considered**
None, I don't know if it is really feasible.

**Comment by Reckless-Satoshi** [comment]
Hey @LeoSpyke 

This is indeed a feasible feature that should be easy to implement.

An straight forward way to go about it is to add 2 new fields to the Order model: `is_private` and `password`. If an order `is_private == True`, it should not be served via `/api/book` . Only a direct link should be able to access the order. Once a user opens the direct link, he should only be able to take it if he knows the password. 

This feature will require both: new backend functionality and new frontend components. I am a bit at capacity lately, but happy to help if anyone wants to take the challenge!

**Comment by Renato1478** [comment]
I think the buyer could have this option as well... I imagined like if you're in a multiplayer game looking for lobbies and there are "public" and "private" matches. The private ones have a password. Is that the idea @LeoSpyke ?

Idk if this is a lot for the escope of the application. I don't see any other DEX having this kind of feature. So maybe is worth collecting users opinions on this just like @Reckless-Satoshi did for the #1154 PR on Telegram.

Anyway, I'm willing to take this enhancement. Assign me if it's still desirable.

**Comment by Reckless-Satoshi** [comment]
> I think the buyer could have this option as well...

Yes, this should be a new field in the `MakerForm` under "Advanced Options" available for both, buy and sell orders.

> Idk if this is a lot for the escope of the application. I don't see any other DEX having this kind of feature. So maybe is worth collecting users opinions on this just like @Reckless-Satoshi did for the #1154 PR on Telegram.

Mmm, I think it is potentially useful! It will be a niche feature though. Therefore, it is best if this option is kind of hidden and does not clutter the UI or makes it difficult to navigate for newbies.

> Anyway, I'm willing to take this enhancement. Assign me if it's still desirable.

Awesome! We can turn it into a Sats rewarded task for 120K Satoshis. I am assigning you.

**Comment by KoalaSat** [comment]
Solved in https://github.com/RoboSats/robosats/pull/2006
